src/xtgeo/grid3d/_egrid.py

```diff
@@ -233,19 +242,34 @@ def to_egrid(self) -> np.ndarray:
         return result
 
 
+def maybe_array_equal(arr1: np.ndarray, arr2: np.ndarray) -> bool:
+    """
+    Returns:
+     True if the two given Optional[np.ndarray]'s are equal (either both None
+     or np.array_equal).
+    """
+    return (arr1 is None and arr2 is None) or np.array_equal(arr1, arr2)
+
+
 @dataclass
-class EGridSubGrid(EclGrid):
+class EGridSubGrid:
```

Comment:
Could I ask the reasoning behind having `EGrid` inherit from `EclGrid`, but remove the inheritance for `EGridSubGrid`? It wasn't obvious for me..

Reply:
So the refactoring changes the behavior from: 1) "`EGrid` contains `EclGrid`s (the global grid and the LGRs)" to 2) "`EGrid` is an `EclGrid`" which is different semantics, but more domain correct. No one is really interested in LGRs as a separate Grid, although in xtgeo the only thing we are capable of is loading the global grid directly and disregard LGRs (we do give a warning though).

I first thought that the 1 approach would be easier, but as it turns out, 2 is actually the way to go.

Reply:
I agree with you!